pkg/test/step.go

```diff
@@ -183,6 +186,32 @@ func (s *Step) Create(namespace string) []error {
 	return errors
 }
 
+// DoKubectl runs all kubectl commands defined for the test step.
+func (s *Step) DoKubectl(namespace string) []error {
+	errs := []error{}
+
+	if s.Step == nil || s.Step.Kubectl == nil {
+		return errs
+	}
+
+	for _, cmd := range s.Step.Kubectl {
+		stdout := &bytes.Buffer{}
+		stderr := &bytes.Buffer{}
+
+		s.Logger.Log("Running kubectl:", cmd)
+
+		err := testutils.Kubectl(context.TODO(), namespace, cmd, s.Dir, stdout, stderr)
+		if err != nil {
+			errs = append(errs, errors.New(stderr.String()))
```

Comment:
I wonder if everything in stderr really means an error. Do you think we should not rather check for return code of the process?

Reply:
Afaik,  err will only be not-nil if exec.Command returns an error (usually exit status). But, you do raise a good point that stderr only gets logged at all if there is an error which may not be desirable.

Reply:
oh I see, we append only if err != nil - 👍